File: app/src/main/java/com/wolo/a222/Cards.java
package com.wolo.a222;

import com.wolo.a222.Model.Firebase.Packs;

import java.util.Arrays;
import java.util.List;
import java.util.Random;


public class Cards {
    private String name;
    private int leftCards;
    private int allcards;
    private List<String> cards;
    private Packs packs;

    public Cards(String name, String[] cards){
        this.name = name;
        this.cards = Arrays.asList(cards);
        this.leftCards = cards.length;
        this.allcards = cards.length;
    }

    public String getName(){
        return name;
    }

    public int sizeCards(){
        return cards.size();
    }

    public String getRandomQuestion(){
        Random number = new Random();
        int r1 = number.nextInt(leftCards);
        String s = cards.get(r1);
        cards.remove(r1);
        return s;

    }

    public void minusOneCard(){
        leftCards = leftCards - 1;
    }

    public String leftCardsText(){
        StringBuilder str = new StringBuilder();
        str.append("Осталось карт в колоде ");
        str.append(cards.size());
        str.append("/");
        str.append(allcards);

        return str.toString();
    }

    public String leftCardsInt(){
        StringBuilder str = new StringBuilder();
        str.append(leftCards);
        str.append("/");
        str.append(allcards);

        return str.toString();
    }

    public List<String> getCards(){
        return cards;
    }

    public void setLeftCards(){
        leftCards -= 1;
    }

    public int getLeftCards(){
        return leftCards;
    }
}

File: app/src/main/java/com/wolo/a222/feature/common/model/Cards.kt
package com.wolo.a222.feature.common.model

import com.wolo.a222.Model.Firebase.Packs
import java.util.*


class Cards(val name: String, cards: ArrayList<String>) {
    var leftCards: Int = 0
        private set
    private val allcards: Int
    private val cards: MutableList<String>
    private val packs: Packs? = null

    val randomQuestion: String
        get() {
            val number = Random()
            val r1 = number.nextInt(leftCards)
            val s = cards[r1]
            cards.removeAt(r1)
            return s

        }

    init {
        this.cards = cards
        this.leftCards = cards.size
        this.allcards = cards.size
    }

    fun sizeCards(): Int {
        return cards.size
    }

    fun minusOneCard() {
        leftCards = leftCards - 1
    }

    fun leftCardsText(): String {
        val str = StringBuilder()
        str.append("Осталось карт в колоде ")
        str.append(cards.size)
        str.append("/")
        str.append(allcards)

        return str.toString()
    }

    fun leftCardsInt(): String {
        val str = StringBuilder()
        str.append(leftCards)
        str.append("/")
        str.append(allcards)

        return str.toString()
    }

    fun getCards(): List<String> {
        return cards
    }

    fun setLeftCards() {
        leftCards -= 1
    }
}
